staging/src/k8s.io/kubectl/pkg/cmd/auth/cani.go

```diff
@@ -416,6 +434,38 @@ func isNamespaced(gvr schema.GroupVersionResource, discoveryClient discovery.Dis
 	return false, fmt.Errorf("the server doesn't have a resource type '%s' in group '%s'", gvr.Resource, gvr.Group)
 }
 
+func hasSubresource(discoveryClient discovery.DiscoveryInterface, gvr schema.GroupVersionResource, subresource string) error {
```

Comment:
Wouldn't be better and easier to do that check in api part instead in here?. This looks to me a complex effort in client side to verify the existence of subresource. Moreover it is a low probability but subresource may not be found due to 6 hours TTL?

Thoughts?

Reply:
SARs are checking only the matching against the rules (RBAC) not existince of a subresource